.github/workflows/unit_tests.yml

```
@@ -0,0 +1,46 @@
+
+jobs:
+  unit_tests:
+    strategy:
+      max-parallel: 3
+      matrix:
+        python-version: [ "3.10", "3.11", "3.12" ]
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v4
+      - name: Set up python ${{ matrix.python-version }}
+        uses: actions/setup-python@v2
```

actionlint (1.7.7)

the runner of "actions/setup-python@v2" action is too old to run on GitHub Actions. update the action's version to fix this issue

(action)